src/megaclient.cpp

```diff
@@ -5543,15 +5720,16 @@ void MegaClient::readtree(JSON* j)
 }
 
 // server-client newnodes processing
-handle MegaClient::sc_newnodes()
+handle MegaClient::sc_newnodes(Node* priorActionpacketDeletedNode, bool& firstHandleMismatchedDelete)
 {
     handle originatingUser = UNDEF;
     for (;;)
     {
         switch (jsonsc.getnameid())
         {
             case 't':
-                readtree(&jsonsc);
+                readtree(&jsonsc, priorActionpacketDeletedNode, firstHandleMismatchedDelete);
+                if (priorActionpacketDeletedNode && firstHandleMismatchedDelete) return originatingUser;
```

Comment:
Not sure, but... would it be possible that the `originatingUser` has not been received (parsed) at this point?

Reply:
Yes, I think you're right.   I think that line was to return early when we detect this case, but in fact it's probably better not to - the return value would only affect useralerts, but we shouldn't skip any other state changes this actionpacket contains.   The critical thing is to detect moves (d followed by t) and treat them as having the same seqtag, even though the t packet can't be tagged with cmdseq, because it would break backward compatibility (due to those hardcoded strings test and test2 that we used to memcmp against).

Reply:
Then... should we use a flag which is set when the `t` packet is received here, so when the parsing completes, we return the valid value?

Reply:
priorActionpacketDeletedNode indicates whether we should be considering if this 't' is the second part of a move, and if it turns out to be readtree() will set firstHandleMatchedDelete true.   It does that by calling readnodes() with those same parameters and that one also returns early without processing anything if it is the case.  
Ok, having re-analyzed this a bit more, I remember what is happening.   This code is replacing those old memcmp() calls that recognised whether the packets were a `d` followed by `t`, both for the same `i` (ie, both a result of the same command).    If that is NOT the case then the `d` the result of an independent command, and we need to exit the procsc() straight away, and let syncdown() run.    When procsc() is called again, to pick up where we left off with the `t` packet, the priorActionpacketDeletedNode will not be set, and the `t` processing won't exit early at any point.      And in the move case, we would not have exited early anyway.

Reply:
Comparing against develop, I see that in fact what it used to do is detect the move, and exit procsc() after both packets of the move were complete (for a run of syncdown()), so I'll submit a PR to this branch to make this adjustment - the idea of exiting after the `d` packet but before the `t` packet was wrong.   Though, if the sync rework works out as intended, we won't need to exit procsc() for this case, but let's keep this branch up to date anyway.